### PR TITLE
AWS/SDK 2.0 and expose ShellCommandRunner errors

### PR DESCRIPTION
**Pull request by apinstein**
@lorennorman trying to expose the underlying exceptions here; this seemed like the easiest way to add the raw exception to the result w/o any BC issues. would love to hear thoughts.

### CODE REVIEW

**Comment by apinstein** [comment]
@jetviper21 you can look too

**Comment by lorennorman** [comment]
Pretty straightforward, this one. If it works, then I'd say it's good to go, but I don't know if it is tested.

File: src/ShellCommand/ShellCommandRunner.php [hunk 1]
```diff
@@ -307,18 +313,10 @@ private function _uploadToS3($localFilePath, $targetUrl)
...
 
     // Upload!
-    $s3       = new AmazonS3($creds);
-    $response = $s3->create_object($bucket, $path, $headers);
-    if (!$response->isOK())
-    {
-      throw new Exception("Upload to S3 {$targetUrl} failed: " . print_r($response->body, true) . PHP_EOL . print_r($response->header, true));
-    }
+    $s3     = S3Client::factory($creds);
+    $result = $s3->upload($bucket . 'sdfjb', $path, $localFilePath);  // throws on error
```
Comment (apinstein) [inline src/ShellCommand/ShellCommandRunner.php]: what really pisses me off is that from the docs I cannot tell if `$result` will ever contain an "error"

Comment (apinstein) [inline src/ShellCommand/ShellCommandRunner.php]: http://docs.aws.amazon.com/aws-sdk-php/latest/class-Aws.S3.S3Client.html#_upload

Comment (apinstein) [inline src/ShellCommand/ShellCommandRunner.php]: it returns a http://docs.aws.amazon.com/aws-sdk-php/latest/class-Guzzle.Service.Resource.Model.html

Guzzle is the HTTP client wrapper. So presumably whether or not the `Model` contains an error is up to the server protocol?

Comment (apinstein) [inline src/ShellCommand/ShellCommandRunner.php]: like maybe from here? http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html

Comment (apinstein) [inline src/ShellCommand/ShellCommandRunner.php]: and it looks like it's documented that all errors are returned thru ["general error codes" http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html]

**Pull request merged**